lib/puppet/provider/service/gentoo.rb

```diff
@@ -10,7 +10,10 @@
 
   commands :update => "/sbin/rc-update"
 
+  # Gentoo supports different init systems, the two most used ones are OpenRC and systemd
+  # OpenRC support is handled via the gentoo provider. It should be the default unless systemd is running
   confine :operatingsystem => :gentoo
+  confine :false => Puppet::FileSystem.exist?('/proc/1/comm') && Puppet::FileSystem.read('/proc/1/comm').include?('systemd')
```

Comment:
I'm not sure if this is the correct syntax/the preferred way. We could also do:
```suggestion
  confine :true => Puppet::FileSystem.exist?('/proc/1/comm') != Puppet::FileSystem.read('/proc/1/comm').include?('systemd')
```

And how are the confines evaluated if we've multiple of them? If that's a boolean or operation this check won't help us.